Cron/CleanupNotifications.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2024 Adyen N.V. (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Cron;
+
+use Adyen\Payment\Api\Repository\AdyenNotificationRepositoryInterface;
+use Adyen\Payment\Cron\Providers\NotificationsProviderInterface;
+use Adyen\Payment\Helper\Config;
+use Adyen\Payment\Logger\AdyenLogger;
+use Adyen\Payment\Model\Notification;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\StoreManagerInterface;
+
+class CleanupNotifications
+{
+    /**
+     * @param NotificationsProviderInterface[] $providers
+     */
+    public function __construct(
+        private readonly array $providers,
+        private readonly AdyenLogger $adyenLogger,
+        private readonly Config $configHelper,
+        private readonly StoreManagerInterface $storeManager,
+        private readonly AdyenNotificationRepositoryInterface $adyenNotificationRepository
+    ) { }
+
+    /**
+     * @return void
+     * @throws NoSuchEntityException
+     */
+    public function execute(): void
+    {
+        $storeId = $this->storeManager->getStore()->getId();
+        $isWebhookCleanupEnabled = $this->configHelper->getIsWebhookCleanupEnabled($storeId);
+
+        if ($isWebhookCleanupEnabled) {
+            $numberOfItemsRemoved = 0;
+
+            foreach ($this->providers as $provider) {
+                /** @var Notification $notificationToCleanup */
+                foreach ($provider->provide() as $notificationToCleanup) {
+                    $isSuccessfullyDeleted = $this->adyenNotificationRepository->delete($notificationToCleanup);
+
+                    if ($isSuccessfullyDeleted) {
+                        $message = __('%1: Notification with entityId %2 has been deleted.',
```

Comment:
Can we add also here one extra info: Notification with entityId %2 has been deleted because it was processed/received X days ago.

Probably means getting that field from the provider

Reply:
Thank you @acampos1916!

You can find the relevant changes in [this](https://github.com/Adyen/adyen-magento2/pull/2837/commits/a8b261d481a9ef616ba95d7a1e0d2ef075da42e6) commit.